test/modern_bpf/event_class/event_class.cpp

```diff
@@ -0,0 +1,267 @@
+	par.len = 0;
+	par.valptr = NULL;
+	m_event_params.push_back(par);
+
+	for(int j = 0; j < nparams; j++)
+	{
+		par.valptr = valptr;
+		par.len = lens16[j];
+		valptr += lens16[j];
+		m_event_params.push_back(par);
+		total_len += lens16[j];
+	}
+
+	/* This event len is the overall len of the event (header + len_vector + data).
+	 * Note: we compute this length according to the number of params written in the header by the bpf program.
+	 */
+	m_event_len = total_len;
+}
+
+/////////////////////////////////
+// GENERIC EVENT ASSERTIONS
+/////////////////////////////////
+
+void event_test::assert_event_presence()
+{
+	uint64_t pid = ::getpid();
+	int consume_ret = 0;
+	uint16_t cpu_id = 0;
+
+	/* We need the while loop because in the buffers there could be different events
+	 * with the type we are searching for. Even if we explicitly create only one event
+	 * of this type, the system could create other events of the same type during the test!
+	 */
+	while(true)
+	{
+		consume_ret = pman_consume_one_from_buffers((void**)&m_event_header, &cpu_id);
+		if(consume_ret == -1 || m_event_header == NULL)
+		{
+			FAIL() << "There is no event in the buffer.";
+			exit(EXIT_SUCCESS);
+		}
+		if(m_event_header->tid == pid && m_event_header->type == m_event_type)
+		{
+			break;
+		}
+	}
+}
+
+void event_test::assert_header()
+{
+	int num_params_from_bpf_table = pman_get_event_params(m_event_type);
+
+	/* the bpf event gets the correct number of parameters from the param table. */
+	ASSERT_EQ(m_event_header->nparams, num_params_from_bpf_table) << "'nparams' in the header is not correct." << std::endl;
+	/* the len specified in the header matches the real event len. */
+	ASSERT_EQ(m_event_header->len, m_event_len) << "'event_len' in the header is not correct." << std::endl;
+}
+
+void event_test::assert_num_params_pushed(int total_params)
+{
+	int num_params_from_bpf_table = pman_get_event_params(m_event_type);
+	ASSERT_EQ(total_params, num_params_from_bpf_table) << "for this event we have not pushed the right number of parameters." << std::endl;
+}
+
+/////////////////////////////////
+// PARAM ASSERTIONS
+/////////////////////////////////
+
+void event_test::assert_empty_param(int param_num)
+{
+	assert_param_boundaries(param_num);
+	/* The param length must be 0. */
+	assert_param_len(0);
+}
+
+void event_test::assert_only_param_len(int param_num, uint16_t expected_size)
+{
+	assert_param_boundaries(param_num);
+	assert_param_len(expected_size);
+}
+
+template <typename T>
+void event_test::assert_numeric_param(int param_num, T param, enum assertion_operators op)
+{
+	assert_param_boundaries(param_num);
+	assert_param_len(sizeof(T));
+
+	switch (op)
+	{
+		case EQUAL:
+		ASSERT_EQ(*(T*)(m_event_params[m_current_param].valptr), param) << VALUE_NOT_CORRECT << m_current_param << std::endl;
+		break;
+
+		case GREATER_EQUAL:
+		ASSERT_GE(*(T*)(m_event_params[m_current_param].valptr), param) << VALUE_NOT_CORRECT << m_current_param << std::endl;
```

Comment:
I think we are missing a `break` here

Reply:
oh yes, great catch!! it is not used right now so I didn't notice it running tests :cry: